frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -112,12 +112,12 @@ export default class Dashboard extends Component {
     });
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (this.props.dashboardId !== nextProps.dashboardId) {
-      this.loadDashboard(nextProps.dashboardId);
+  componentDidUpdate(prevProps) {
```

Comment:
`UNSAFE_componentWillReceiveProps` was causing a race condition, actually, where `this.props.dashboardId` had changed but `this.props.location` had not.